### PR TITLE
Simplify installing tracers

### PR DESCRIPTION
**Pull request by nbroyles**
Removes a superfluous tracer field and adds documentation to Jaeger configuration classes.

### CODE REVIEW

File: misk/src/main/kotlin/misk/tracing/backends/jaeger/JaegarSamplerConfig.kt [hunk 1]
```diff
@@ -2,6 +2,10 @@ package misk.tracing.backends.jaeger
 
 import misk.config.Config
 
+/**
+ * Configuration for Jaeger's sampler. If values are left null then Jaeger will provide defaults.
+ * See [com.uber.jaeger.Configuration.SamplerConfiguration] for default values.
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/tracing/backends/jaeger/JaegarSamplerConfig.kt]: are the default values public? if so, might be nice to inline and remove the nullability

File: misk/src/main/kotlin/misk/tracing/TracingModule.kt [hunk 2]
```diff
@@ -5,12 +5,12 @@ import misk.tracing.backends.jaeger.JaegerBackendModule
...
   override fun configure() {
-    if (config.tracer.equals("jaeger", true)) {
+    if (config.backends.jaeger != null) {
       install(JaegerBackendModule(config.backends.jaeger))
       return
     }
 
-    throw IllegalArgumentException("No backend for '${config.tracer}' tracer. Ensure that " +
-        "tracer name is spelled correctly and that a backend module exists for that tracer.")
+    throw IllegalStateException("No backends configured for tracing. Please update your yaml "
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/tracing/TracingModule.kt]: basically the tab vs. whitespace debate of exception throwing. I see you've chosen your religion

File: misk/src/main/kotlin/misk/tracing/backends/jaeger/JaegarSamplerConfig.kt (hunk 1 above)
Comment (nbroyles) [reply misk/src/main/kotlin/misk/tracing/backends/jaeger/JaegarSamplerConfig.kt]: Not all of them which is super annoying. That was my initial plan until I noticed difference levels of visibility. 

Also would've been nice if they had public builders for their configs too which would have made instantiating them a little less gross.

:-/

File: misk/src/main/kotlin/misk/tracing/TracingModule.kt (hunk 2 above)
Comment (nbroyles) [reply misk/src/main/kotlin/misk/tracing/TracingModule.kt]: 😬

**Pull request merged**